transaction/std/src/transaction_builder.rs

```diff
@@ -620,19 +650,70 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
             .map(TryInto::try_into)
             .collect::<Result<Vec<InputRing>, _>>()?;
 
-        let message = tx_prefix.hash().0;
+        Ok(UnsignedTx {
+            tx_prefix,
+            rings: input_rings,
+            output_secrets,
+            block_version: self.block_version,
+        })
+    }
+
+    /// Consume the builder and return the transaction.
+    pub fn build<RNG: CryptoRng + RngCore, S: RingSigner + ?Sized>(
+        self,
+        ring_signer: &S,
+        rng: &mut RNG,
+    ) -> Result<Tx, TxBuilderError> {
+        self.build_with_comparer_internal::<RNG, DefaultTxOutputsOrdering, S>(ring_signer, rng)
+    }
+
+    /// Consume the builder and return the transaction with a comparer.
+    /// Used only in testing library.
+    #[cfg(feature = "test-only")]
+    pub fn build_with_sorter<
+        RNG: CryptoRng + RngCore,
+        O: TxOutputsOrdering,
+        S: RingSigner + ?Sized,
+    >(
+        self,
+        ring_signer: &S,
+        rng: &mut RNG,
+    ) -> Result<Tx, TxBuilderError> {
+        self.build_with_comparer_internal::<RNG, O, S>(ring_signer, rng)
+    }
+
+    /// Consume the builder and return the transaction with a comparer
+    /// (internal usage only).
+    fn build_with_comparer_internal<
+        RNG: CryptoRng + RngCore,
+        O: TxOutputsOrdering,
+        S: RingSigner + ?Sized,
+    >(
+        self,
+        ring_signer: &S,
+        rng: &mut RNG,
+    ) -> Result<Tx, TxBuilderError> {
+        // TODO Maybe include these inside TransactionSigningData?
+        let block_version = self.block_version;
```

Comment:
If we put the block_version and fee inside `UnsignedTx`, we can then have a `sign` method on `UnsignedTx`, and this function becomes two lines:
```
let unsigned_tx = self.build_unsigned::<RNG, O>()?;
unsigned_tx.sign(ring_signer, rng)
```

Reply:
Actually looks like the fee is already inside `UnsignedTx`, and so is the `block_version`. So this can just use `.sign()`